sandbox/apparmor/apparmor.go

```diff
@@ -316,43 +321,92 @@ func probeKernelFeatures() ([]string, error) {
 }
 
 func probeParserFeatures() ([]string, error) {
-	parser, err := findAppArmorParser()
+	parser, args, internal, err := FindAppArmorParser()
 	if err != nil {
 		return []string{}, err
 	}
-	features := make([]string, 0, 4)
-	if tryAppArmorParserFeature(parser, "change_profile unsafe /**,") {
-		features = append(features, "unsafe")
-	}
-	if tryAppArmorParserFeature(parser, "network qipcrtr dgram,") {
-		features = append(features, "qipcrtr-socket")
-	}
-	if tryAppArmorParserFeature(parser, "capability bpf,") {
-		features = append(features, "cap-bpf")
+	var featureProbes = []struct {
+		feature string
+		probe   string
+	}{
+		{
+			feature: "unsafe",
+			probe:   "change_profile unsafe /**,",
+		},
+		{
+			feature: "include-if-exists",
+			probe:   `#include if exists "/foo"`,
+		},
+		{
+			feature: "qipcrtr-socket",
+			probe:   "network qipcrtr dgram,",
+		},
+		{
+			feature: "cap-bpf",
+			probe:   "capability bpf,",
+		},
+		{
+			feature: "cap-audit-read",
+			probe:   "capability audit_read,",
+		},
+	}
+	features := make([]string, 0, 5)
+	for _, fp := range featureProbes {
+		if tryAppArmorParserFeature(parser, args, fp.probe) {
+			features = append(features, fp.feature)
+		}
 	}
-	if tryAppArmorParserFeature(parser, "capability audit_read,") {
-		features = append(features, "cap-audit-read")
+	if internal {
+		features = append(features, "snapd-internal")
 	}
 	sort.Strings(features)
 	return features, nil
 }
 
-// findAppArmorParser returns the path of the apparmor_parser binary if one is found.
-func findAppArmorParser() (string, error) {
+// FindAppArmorParser returns the path of the apparmor_parser binary if one
+// is found as well as any required command-line arguments to use when
+// invoking the binary, and a boolean to indicate whether this is internal
+// to snapd (ie is provided by snapd)
+func FindAppArmorParser() (path string, args []string, internal bool, err error) {
+	// first see if we have our own internal copy which could come from
+	// the snapd snap (likely) or be part of the snapd distro package (unlikely)
+	args = make([]string, 0)
+	if path, err := snapdtool.InternalToolPath("apparmor_parser"); err == nil {
+		if osutil.IsExecutable(path) {
+			prefix := strings.TrimSuffix(path, "apparmor_parser")
+			// when using the internal apparmor_parser also use
+			// it's own configuration and includes etc plus
```

Comment:
super nitpick: its

Reply:
Good point but this is in the original commit which I would rather restore with a single "Revert" commit so I don't think it's worth fixing at this point with an extra commit.